app/components/user-profile-ics.js

```diff
@@ -35,17 +35,32 @@ export default class UserProfileIcsComponent extends Component {
   /**
    * Generate a random token from a combination of
    * the user id, a random string and the current time
-   * Implementation lifted from https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest
-   * @param userId
-   * @return String
+   *
+   * We use two different methods for making this work. Because some versions of safari don't allow
```

Comment:
regarding the Safari compatibility issue  - please add a link to https://developer.mozilla.org/en-US/docs/Web/API/Crypto/subtle#browser_compatibility in the comments, since it shows the problem in the compat matrix.

Reply:
Good call, handled ✅